docs/en/integrations/system-testing.asciidoc

```diff
@@ -0,0 +1,280 @@
+
+[source,terminal]
+----
+<package root>/
+  data_stream/
+    <data stream>/
+      manifest.yml
+  manifest.yml
+----
+
+To define a system test we must define configuration on at least one level: a package or a data stream's one.
+
+First, we must define the configuration for deploying a package's integration service. We can define it on either the package level:
+
+[source,terminal]
+----
+<package root>/
+  _dev/
+    deploy/
+      <service deployer>/
+        <service deployer files>
+----
+
+or the data stream's level:
+
+[source,terminal]
+----
+<package root>/
+  data_stream/
+    <data stream>/
+      _dev/
+        deploy/
+          <service deployer>/
+            <service deployer files>
+----
+
+`<service deployer>` - a name of the supported service deployer:
+
+* `docker` - Docker Compose
+* `k8s` - Kubernetes
+* `tf` - Terraform
+
+[discrete]
+[[system-docker-compose]]
+=== Docker Compose service deployer
+
+When using the Docker Compose service deployer, the `<service deployer files>` must include a `docker-compose.yml` file.
+The `docker-compose.yml` file defines the integration service(s) for the package. If your package has a logs data stream,
+the log files from your package's integration service must be written to a volume. For example, the `apache` package has
+the following definition in it's integration service's `docker-compose.yml` file.
+
+[source,terminal]
+----
+version: '2.3'
+services:
+  apache:
+    # Other properties such as build, ports, etc.
+    volumes:
+      - ${SERVICE_LOGS_DIR}:/usr/local/apache2/logs
+----
+
+Here, `SERVICE_LOGS_DIR` is a special keyword. It is something that we will need later.
+
+[discrete]
+[[system-terraform]]
+=== Terraform service deployer
+
+When using the Terraform service deployer, the `<service deployer files>` must include at least one `*.tf` file.
+The `*.tf` files define the infrastructure using the Terraform syntax. The terraform based service can be handy to boot up
+resources using selected cloud provider and use them for testing (e.g. observe and collect metrics).
+
+Sample `main.tf` definition:
+
+[source,terminal]
+----
+variable "TEST_RUN_ID" {
+  default = "detached"
+}
+
+provider "aws" {}
+
+resource "aws_instance" "i" {
+  ami           = data.aws_ami.latest-amzn.id
+  monitoring = true
+  instance_type = "t1.micro"
+  tags = {
+    Name = "elastic-package-test-${var.TEST_RUN_ID}"
+  }
+}
+
+data "aws_ami" "latest-amzn" {
+  most_recent = true
+  owners = [ "amazon" ] # AWS
+  filter {
+    name   = "name"
+    values = ["amzn2-ami-hvm-*"]
+  }
+}
+----
+
+Notice the use of the `TEST_RUN_ID` variable. It contains a unique ID, which can help differentiate resources created in potential concurrent test runs.
+
+[discrete]
+[[system-kubernetes]]
+=== Kubernetes service deployer
+
+The Kubernetes service deployer requires the `_dev/deploy/k8s` directory to be present. It can include additional `*.yaml` files to deploy
```

Comment:
Same concern about service deployers. Does it mean that we have to apply changes in both places to keep them up-to-date?

Reply:
It's a good point. I'll bring this up in our weekly tomorrow. There might just be too many manual copy/paste steps required to move forward with this PR at this point.

Reply:
In the end, I may just have to updated things manually for a release or two until we figure out automation